src/librustc_metadata/decoder.rs

```diff
@@ -1365,6 +1366,30 @@ impl<'a, 'tcx> CrateMetadata {
         // This shouldn't borrow twice, but there is no way to downgrade RefMut to Ref.
         self.source_map_import_info.borrow()
     }
+
+    /// Get the `DepNodeIndex` corresponding this crate. The result of this
+    /// method is cached in the `dep_node_index` field.
+    pub(super) fn get_crate_dep_node_index(&self, tcx: TyCtxt<'tcx>) -> DepNodeIndex {
```

Comment:
We're somewhat clearly doing this with an already-existing TyCtxt, so maybe it would be simpler for this to be a query (if, admittedly, less efficient?)

Reply:
Ah, right. Yeah that makes sense.